golem-worker-service-base/src/worker_bridge_execution/content_type_mapper.rs

```diff
@@ -0,0 +1,938 @@
+use golem_wasm_ast::analysis::AnalysedType;
+use golem_wasm_rpc::TypeAnnotatedValue;
+use poem::web::headers::ContentType;
+use poem::web::WithContentType;
+use poem::Body;
+use std::fmt::{Display, Formatter};
+
+pub trait HttpContentTypeResponseMapper {
+    fn to_http_response_body(
+        &self,
+        content_type_headers: ContentTypeHeaders,
+    ) -> Result<WithContentType<Body>, ContentTypeMapError>;
+}
+
+pub enum ContentTypeHeaders {
+    FromClientAccept(Vec<ContentType>),
+    FromUserDefinedResponseMapping(ContentType),
+    Empty,
+}
+
+impl ContentTypeHeaders {
+    pub fn from(
+        response_content_type: Option<ContentType>,
+        accepted_content_types: Vec<ContentType>,
+    ) -> Self {
+        if let Some(response_content_type) = response_content_type {
+            ContentTypeHeaders::FromUserDefinedResponseMapping(response_content_type)
+        } else if !accepted_content_types.is_empty() {
+            ContentTypeHeaders::FromClientAccept(accepted_content_types)
+        } else {
+            ContentTypeHeaders::Empty
+        }
+    }
+}
+
+impl HttpContentTypeResponseMapper for TypeAnnotatedValue {
+    fn to_http_response_body(
+        &self,
+        content_type_headers: ContentTypeHeaders,
+    ) -> Result<WithContentType<Body>, ContentTypeMapError> {
+        match content_type_headers {
+            ContentTypeHeaders::FromUserDefinedResponseMapping(content_type) => {
+                internal::get_body_from_response_content_type(self, &content_type)
```

Comment:
do we need two functions ?

in general i think it should be possible to use `get_response_body_from_accept_content_headers`
where theres is only one content type

Reply:
Which one will you pick from the accept headers?

Reply:
Anyway I will try to squash them into 1. But I am not a big fan of squashing it somehow in the naem of reusability. Accept HEADERS can be multiple in value, the response header can depend on the priority oif that list based on quality parameters, and in future we may have things that divert from logic where it handles only a single ContentType. Note that AcceptHeaders is not a List<ContentType> either